### PR TITLE
refactor to prepare for struct logs

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

refactor flags.go parseFlags to prepare for struct logs.

We are looking to split the flag parsing from the validation as we want to pass the logger to the validation function.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: cmd/nginx-ingress/flags.go [hunk 1]
```diff
@@ -402,6 +337,73 @@ func mustValidateFlags() {
...
+
+	if *enableLatencyMetrics && !*enablePrometheusMetrics {
+		glog.Warning("enable-latency-metrics flag requires enable-prometheus-metrics, latency metrics will not be collected")
+		*enableLatencyMetrics = false
+	}
+
+	if *enableServiceInsight && !*nginxPlus {
+		glog.Warning("enable-service-insight flag support is for NGINX Plus, service insight endpoint will not be exposed")
+		*enableServiceInsight = false
+	}
```
Comment (pdabelf5) [inline cmd/nginx-ingress/flags.go:385]: Should these warning checks be in `mustValidateInitialChecks()`?

File: cmd/nginx-ingress/flags.go [hunk 2]
```diff
@@ -220,77 +220,12 @@ func parseFlags() {
 	if *versionFlag {
```
Comment (j1m-ryan) [inline cmd/nginx-ingress/flags.go:220]: should this `versionFlag` be in initValidation?
also do we still need the `parseFlags` function in addition to `mustValidateFlags`?

File: cmd/nginx-ingress/flags.go [hunk 3]
```diff
@@ -220,77 +220,12 @@ func parseFlags() {
...
 		os.Exit(0)
 	}
+}
 
+func initValidation() {
```
Comment (j1m-ryan) [inline cmd/nginx-ingress/flags.go:225]: should we call this `mustValidate` or something similar?

**Review by pdabelf5 — APPROVED** [review]

**Review by jjngx — APPROVED** [review]
🚀

**Review by j1m-ryan — APPROVED** [review]

**Pull request merged**